applications/array-optimization.ipynb

```diff
@@ -0,0 +1,240 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Array Scheduling Optimization\n",
+    "\n",
+    "In this example, we'll perform a typical workflow\n",
+    "\n",
+    "* Load batches of data\n",
+    "* Stack batches into a single Dask Array\n",
+    "* Rechunk the data for downstream processing\n",
+    "* Write rechunked data to disk, using [Zarr](https://zarr.readthedocs.io)\n",
+    "\n",
+    "Depending on the rechunking pattern, this is an embarassingly parallel opertion. However,\n",
+    "the Dask scheduler doesn't necessarily know that, which may result in sub-optimal scheduling. We'll configure the optimization settings to ensure that we have ideal scheduling to minimize data transfer across the cluster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!conda install -y -c conda-forge zarr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask.distributed import Client\n",
+    "import dask.array as da\n",
+    "import zarr\n",
+    "\n",
+    "client = Client(processes=False, threads_per_worker=4,\n",
+    "                n_workers=1, memory_limit='2GB')\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Rather than loading data from disk, we'll generate random data in memory.\n",
+    "We'll have 25 batches, each of which has shape `(2,000,000 x 90,000)`.\n",
+    "\n",
+    "We rechunk along just the first axis (preserving the chunking on the second axis)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inputs = [da.random.random(size=2_000_000, chunks=90_000)\n",
+    "          for _ in range(25)]\n",
```

Comment:
Minor point, but it might be good to increase the number of inputs from 25 to something greater than 50 (e.g., 100), and then also increase the fuse_ave_width to 50, so that the connection is clear between the value of fuse_ave_width and the size of the chunks in the rechunk operation (and not the number of inputs).